engine/script/src/test/test_json.lua

```diff
@@ -57,7 +57,9 @@ function test_json_invalid_primitive()
 end
 
 function test_json_valid_primitive()
-    assert(json.decode("null") == json.null)
+    assert(json.decode("null") == nil)
+    assert(json.decode("null", { decode_null_as_userdata = false }) == nil)
+    assert(json.decode("null", { decode_null_as_userdata = true }) == json.null)
```

Comment:
Can we add a test for testing the regression?
I.e that the "key":null shouldn't become "key":json.null ?

Reply:
Added!